### ISSUE TITLE
Wymagania prawa człowieka - requirements from international civil society

### ISSUE THREAD

**Issue by broukema** [ProteGO-app/specs#37]
Here is the document prepared by multiple international human rights groups on requirements from governments and from software such as this:
https://www.amnesty.org/download/Documents/POL3020812020ENGLISH.pdf

There are eight requirements listed. The technical aspects that I see are:
* points 2 and 3 - "timebound" - make sure that the tokens are destroyed after, say 21 days?
* point 4 - data security, anonymity: by free-licensing the code and distributing it on git repositories, this is likely to be satisfied by this project

The other points seem mostly political/administrative tasks - people developing the software can at the most remind GIS that Amnesty International, Human Rights Watch, and about 100 or so smaller human rights and digital rights community groups want this to be done properly, not North Korean/Microsoft style.

**Comment by AlexF0rtune** [comment]
As an Ex-Microsoftee, I protest ;)

Now, as far as I know, there is indeed a time-limit on holding the data - the assumption is that all data older than 14 days must be purged. Furthermore, no information about encounters shall be pushed remotely unless one is found out to be infected.

**Comment by potiuk** [comment]
And the data should not be tied to phone number as it is now. this is unacceptable violation of civil rights.

**Comment by potiuk** [comment]
> Now, as far as I know, there is indeed a time-limit on holding the data - the assumption is that all data older than 14 days must be purged. Furthermore, no information about encounters shall be pushed remotely unless one is found out to be infected.

Not really. Even if the data is not pushed directly by non-infected it is in fact pushed by the infected. The infected people act as rely to push the data from everyone else - the IDs of other people are pseudo-anonymous and they can be de-anonymised and tied to mobile numbers on the server side.

See #34 for details and description of good, privacy-aware way of implementing it proposed by scientists.

**Comment by AlexF0rtune** [comment]
> the IDs of other people are pseudo-anonymous and they can be de-anonymised and tied to mobile numbers on the server side.

Indeed, and I understood this to be a requirement, perhaps less the tying to mobile numbers?

**Comment by potiuk** [comment]
What do you mean ? Could you please elaborate

**Comment by potiuk** [comment]
Be careful everyone in this thread - I had almost been a victim of targetted phising attack right after I started to comment here. If you got an email from Github stating that there was a suspicious activity on your account - take a look at the sender and link that you get.

**Comment by jakublipinski** [comment]
Please indeed be careful. I noticed a mysterious user added to this GitHub org. I never added this user. I cannot see the event related to this user in the org event log. I contacted GitHub and asked them for explanation. We might be subject to some sabotage.

**Comment by jakublipinski** [comment]
Thanks @broukema. Points 2 and 3 ("timebound" ids) are not yet implemented in our project. They will be. I leave this issue open until it's done.

**Comment by broukema** [comment]
There are some more technical declarations of privacy principles floating around too:
* https://www.ccc.de/en/updates/2020/contact-tracing-requirements
* https://github.com/TCNCoalition/TCN/blob/main/ContactTracingBillOfRights.md
* https://law.mit.edu/pub/covid19contacttracingprivacyprinciples

I guess that it won't take long before there's an [en.Wikipedia "comparison of" type page](https://en.wikipedia.org/wiki/Comparison_of_microblogging_services)
 comparing the various ongoing projects for their various characteristics; here is a list including this project (for easy copy/pasting):
* https://github.com/ProteGO-app
* https://github.com/DP-3T/documents
* https://github.com/TCNCoalition/TCN
As long as all the projects are free-licensed, these should all accelerate each other's development, so keep an eye on the other two for code under compatible licences that can be re-used or re-factored here or discussions of design strategy and dealing with difficult issues. :)
